src/Symfony/Component/HttpClient/Response/AsyncResponse.php

```diff
@@ -95,7 +95,7 @@ public function toStream(bool $throw = true)
         }
 
         $handle = function () {
-            $stream = StreamWrapper::createResource($this->response);
+            $stream = $this->response instanceof StreamableInterface ? $this->response->toStream(false) : StreamWrapper::createResource($this->response);
```

Comment:
wouldn't this bypass changes done by the passthru callable ?

Reply:
nope: this only extracts any potential handle for stream_select(), but the stream is always read through the resource created L103 below, which goes through the AsyncResponse